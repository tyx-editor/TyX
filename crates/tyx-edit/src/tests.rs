use std::path::PathBuf;

use super::*;

/// Makes snapshot tests for the given function.
/// The fixtures used for snapshot testing are in the `src/fixtures/{name}`
/// directory.
fn snapshot_testing(name: &str, f: &impl Fn(LspWorld, PathBuf)) {
    tinymist_tests::snapshot_testing!(name, |verse, path| {
        f(verse.snapshot(), path);
    });
}

#[test]
fn test_preview() {
    snapshot_testing("integration", &|world, _path| {
        let result = preview(&world).unwrap_or_else(|| "null".into());
        let hash = tinymist_std::hash::hash128(&result);
        insta::with_settings!({
            description => format!("siphash128_13: {hash:016x}"),
        }, {
            insta::assert_binary_snapshot!(".html", result.into_bytes());
        })
    });
}
