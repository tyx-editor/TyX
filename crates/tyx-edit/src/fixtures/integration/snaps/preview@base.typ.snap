---
source: crates/tyx-edit/src/tests.rs
description: "siphash128_13: 79b6c2afd5aaf8ebcc549a237dd359ed"
expression: result.into_bytes()
input_file: crates/tyx-edit/src/fixtures/integration/base.typ
extension: html
snapshot_kind: binary
---
